client/ClientDecisionServiceTest/LocalModeTest.cs

```diff
@@ -103,24 +105,140 @@ public void TestDSLocalModelUpdate()
             string guid1 = Guid.NewGuid().ToString();
             string guid2 = Guid.NewGuid().ToString();
             byte[] prevModel = null;
+            int action;
             
             // Generate interactions and ensure the model updates at the right frequency
             // (updates every example initially)
             prevModel = dsLocal.Model;
-            dsLocal.ChooseAction(guid1, context, 1);
+            dsLocal.ChooseActionAsync(guid1, context, 1);

```

Comment:
I think you should await this

Reply:
you can make the test method async

Reply:
Good catch. I actually need to obtain the result synchronously because otherwise the attempt to report rewards and complete the event might not find anything there to complete.